### PR TITLE
Add IScriptServiceV3Alpha contract

### PR DESCRIPTION
**Pull request by APErebus**
# Background

To support the Kubernetes Agent, which will be adding the ability to execute scripts via Kubernetes Jobs, we need to change the `StartScriptCommand` contract. As a result, we have decided to create a new `IScriptServiceV3Alpha`.

# Results

The main requirement for the contract change is that we need information from Server when running scripts as k8s Jobs. We need to know the container image and any external feed credentials/url.

To support this, I've added a new `IScriptExecutionContext ExecutionContext { get; }`  property to the `StartScriptCommandV3Alpha`.
There is one implementation of this interface, `LocalShellScriptExecutionContext` for the existing execution process (a local shell). In a future PR, there will be a `KubernetesJobScriptExecutionContext` (or something similiar). I didn't version this interface as the namespace it's in is versioned.

A future PR will be the initial implementation of this service, which will be a direct copy of the V2 service initially.

Shortcut story: [sc-62810]

# How to review this PR

Quality :heavy_check_mark:

# Pre-requisites

- [ ] I have read [How we use GitHub Issues](https://github.com/OctopusDeploy/Issues/blob/master/docs/CONTRIBUTING.internal.md) for help deciding when and where it's appropriate to make an issue.
- [ ] I have considered informing or consulting the right people, according to the [ownership map](https://whimsical.com/ownership-map-NzbiD4HJyvhC9jNJNfS6TG).
- [ ] I have considered appropriate testing for my change.

### CODE REVIEW

File: source/Octopus.Tentacle.Contracts/ScriptServiceV3Alpha/ScriptCommandV3Alpha.cs [hunk 1]
```diff
@@ -0,0 +1,12 @@
+﻿namespace Octopus.Tentacle.Contracts.ScriptServiceV3Alpha
+{
+    public abstract class ScriptCommandV3Alpha
```
Comment (APErebus) [inline source/Octopus.Tentacle.Contracts/ScriptServiceV3Alpha/ScriptCommandV3Alpha.cs:3]: I create a base command to enforce that all commands must have the ScriptTicket property

File: source/Octopus.Tentacle.Contracts/ScriptServiceV3Alpha/StartScriptCommandV3Alpha.cs [hunk 2]
```diff
@@ -0,0 +1,100 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Newtonsoft.Json;
+
+namespace Octopus.Tentacle.Contracts.ScriptServiceV3Alpha
+{
+    public class StartScriptCommandV3Alpha : ScriptCommandV3Alpha
```
Comment (sburmanoctopus) [inline source/Octopus.Tentacle.Contracts/ScriptServiceV3Alpha/StartScriptCommandV3Alpha.cs:8]: Just thinking out aloud, how much do we want to force customers to specify the `scriptExecutionContext`?

To start with the K8 version will be very rarely used. 

Even when it takes off any everyone thinks it's amazing, I would guess the local shell context is more commonly used.

This makes me wonder, should we make it easier to create these types of commands somehow?

Either by 
- making a constructor that doesn't take the context, and then default it to the local shell context
- some factory style methods?

We don't have to do any of this, just some thoughts 🙂

**Review by sburmanoctopus — APPROVED** [review]
LGTM

File: source/Octopus.Tentacle.Contracts/ScriptServiceV3Alpha/ScriptCommandV3Alpha.cs (hunk 1 above)
Comment (APErebus) [reply source/Octopus.Tentacle.Contracts/ScriptServiceV3Alpha/ScriptCommandV3Alpha.cs:3]: Yeah, that's fine. I don't feel strongly about it, so happy to defer.

File: source/Octopus.Tentacle.Contracts/ScriptServiceV3Alpha/StartScriptCommandV3Alpha.cs (hunk 2 above)
Comment (APErebus) [reply source/Octopus.Tentacle.Contracts/ScriptServiceV3Alpha/StartScriptCommandV3Alpha.cs]: I believe in the server codebase we have some builders that are used for this purpose. In this case, we would default the context to the local shell execution, and then have methods for setting the required k8s context information, which would then build it up.

File: source/Octopus.Tentacle.Contracts/ScriptServiceV3Alpha/ScriptCommandV3Alpha.cs (hunk 1 above)
Comment (APErebus) [reply source/Octopus.Tentacle.Contracts/ScriptServiceV3Alpha/ScriptCommandV3Alpha.cs]: Addressed in c679e679

**Pull request merged**